tests/scripts/github_mergebot.py

```diff
@@ -0,0 +1,552 @@
+            return True
+
+        logging.info(f"Comment {comment} was not from author or committers and is not mergable")
+        return False
+
+    def merge_requested(self) -> bool:
+        """
+        Check if this PR has had a merge requested
+        """
+        merge_commands = [
+            "merge",
+            "merge this",
+            "merge this pr",
+        ]
+        cancel_commands = [
+            "cancel",
+            "cancel merge",
+            "cancel the merge",
+            "stop",
+            "stop merge",
+            "stop the merge",
+        ]
+
+        def parse_action(comment: Dict[str, Any]) -> Optional[str]:
+            if not self.comment_can_merge(comment):
+                return None
+
+            body = comment["body"]
+            if any(f"@tvm-bot {c}" in body for c in merge_commands):
+                return "merge"
+
+            if any(f"@tvm-bot {c}" in body for c in cancel_commands):
+                return "cancel"
+
+            return None
+
+        # Check regular comments
+        all_comments = []
+        for comment in self.raw["comments"]["nodes"]:
+            all_comments.append(comment)
+
+        # Check top-level review comments
+        for review in self.reviews():
+            all_comments.append(review)
+
+        all_comments = sorted(all_comments, key=lambda comment: comment["updatedAt"])
+        actions = [parse_action(comment) for comment in all_comments]
+        logging.info(f"Found these tvm-bot actions: {actions}")
+        actions = [a for a in actions if a is not None]
+
+        if len(actions) == 0:
+            return False
+
+        return actions[-1] == "merge"
+
+    def merge_if_passed_checks(self):
+        # NEUTRAL is GitHub Action's way of saying cancelled
+        failed_ci_jobs = [
+            job
+            for job in self.ci_jobs()
+            if job["status"] not in {"SUCCESS", "SUCCESSFUL", "NEUTRAL", "SKIPPED"}
+        ]
+        all_ci_passed = False
+        has_one_approval = False
+        if len(failed_ci_jobs) > 0:
+            failed_jobs_msg = "\n".join(
+                [f" * [{job['name']} (`{job['status']}`)]({job['url']})" for job in failed_ci_jobs]
+            )
+            self.comment(
+                f"Cannot merge, these CI jobs are not successful on {self.head_oid()}:\n{failed_jobs_msg}"
+            )
+            return
+        else:
+            all_ci_passed = True
+
+        # Map of job name: has seen in completed jobs
+        seen_expected_jobs = {
+            "tvm-ci/pr-merge": False,
+        }
+        logging.info(f"Expected to see jobs: {seen_expected_jobs}")
+
+        missing_expected_jobs = []
+        for job in self.ci_jobs():
+            seen_expected_jobs[job["name"]] = True
+
+        for name, seen in seen_expected_jobs.items():
+            if not seen:
+                missing_expected_jobs.append(name)
```

Comment:
Can split the CI logic out into its own function to keep `merge_if_passed_checks` just checking the various statuses?